### DISCUSSION TITLE
Visual Studio 2022 - GitHub Desktop - Authors Implementation

### DISCUSSION THREAD

**Discussion by fomustangrd88** [community/community#120735] [Programming Help]
### Body

New to incorporating GitHub Desktop with a corporate setting versus personal machine. At work moving off of Microsoft Visual SourceSafe (don't laugh) since upgrading to VS 2022 from 2019. With Visual SourceSafe there's a setting that "Uses network name for automatic user login." Since I actually can't test yet on work machine since still working on creating requirements documents for our upgrade can anyone tell me does this automatically happen when checking-in/out from Visual Studio/GitHub Desktop or is there a setting? All of us developers launch our Visual Studio instance to run as "admin", which will be using our CAC card as validation (i.e., testuser-w)? Several users will be checking in and out amongst all of our different repositories and should each be separate "authors" using that admin username mentioned earlier.

Thanks!

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by david-wei-01001** [comment]
Incorporating GitHub with Visual Studio and GitHub Desktop in a corporate setting, especially when transitioning from a system like Microsoft Visual SourceSafe, involves a few considerations regarding user identification and management.

When you use Git, the commits you make are identified by an author name and email address. This identification is configured locally for each Git environment and is not inherently tied to the network name or the user's login credentials as with Visual SourceSafe.

Here are the key points to manage author identification in Git through Visual Studio and GitHub Desktop:

### Configuring Git Author Identity
1. **Global Configuration**: Each developer can set their Git username and email globally on their machine using the Git command-line interface with the following commands:
   ```bash
   git config --global user.name "Your Name"
   git config --global user.email "your.email@example.com"
   ```
   This information will be used for all Git operations on that machine and will appear as the author of the commits.

2. **Local Repository Configuration**: Developers can also set their username and email on a per-repository basis by running the same commands without the `--global` flag inside the repository directory. This overrides the global settings for that specific repository.

3. **GitHub Desktop**: In GitHub Desktop, you can set the name and email under `Preferences` > `Git`. This is particularly useful if the email used for commits needs to be the same as the one used for the GitHub account to ensure proper linking of commit activities.

4. **Visual Studio**: Visual Studio typically uses the global Git settings but also allows you to configure the user for commits per-repository. This can be set in the repository settings within Team Explorer.

### Corporate Environment Considerations
1. **Running as Admin**: Running Visual Studio as an admin with CAC card validation shouldn't impact the Git author settings as these are separate from the Windows user credentials.

2. **CAC Cards and Smart Cards**: If your corporate environment uses CAC cards for authentication, you'll need to ensure that your Git client is set up to work with these cards for operations that require authentication, such as pushing to a remote repository.

3. **Network Names**: Git does not automatically use network names for commit author identification. You'll need to configure the Git author identity separately as mentioned above.

4. **Managing Multiple Users**: If multiple users are sharing a machine and need to commit with their own identities, each user should configure their Git username and email before committing.

5. **Credential Management**: For operations that require GitHub credentials, such as pushing or fetching from remote repositories, you can use the Git Credential Manager, which integrates with Windows Authentication and can handle tokens, two-factor authentication, and other authentication methods.

6. **Security Policies**: Check with your IT department for any corporate policies regarding the use of Git and GitHub, as there may be specific configurations or constraints you'll need to follow.

In summary, while Git does not automatically use network names for commit author identification, you can set the author information locally for each developer. For authentication-related operations, you'll need to make sure Git is set up to work with your corporate authentication mechanisms, such as CAC cards.

**Comment by fomustangrd88** [comment]
Thanks for all that! I was put in charge of doing our upgrade from VS 2019
to VS 2022, so obviously I need to make sure I understand the possibilities
and do it correctly.

So, if each developer will have GitHub Desktop on our own local machines,
but our code baseline lives on a file share for which all of us have a
mapped drive to and we will not be putting ANYTHING in a GitHub cloud
(everything local), does everything you mentioned above apply the same?
Should I look into credential management? Since nothing "automatically"
gets pulled down (i.e., network name or the admin account name VS is run as
via my CAC card) each developer will run that --global command and manually
add in their author information?

Thanks again!


-----------------------------------

Jason Vaughan

Web Developer

IG @theturfnerd

301.514.6812

**Comment by fomustangrd88** [comment]
Did anyone else have any input? My product request has now been granted and ready for implementation. Thanks!
